### ISSUE TITLE
Failed to verify app state - got invalid redirectURL \"/\"

### ISSUE THREAD

**Issue by jmmclean** [argoproj/argo-cd#10058]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**
After upgrading from 2.4.4 to 2.4.7 we received intermittent errors with the below message
```
argocd-server-5c58f5c844-cksjw argocd-server time="2022-07-20T15:55:59Z" level=warning msg="Failed to verify app state - got invalid redirectURL \"/\""
```

**To Reproduce**
1. Have valid okta configuration for SSO (validate w/ 2.4.4)
2. upgrade from 2.4.4 to 2.4.7
3. some users can log in, some cant (weird)

**Expected behavior**
Argo CD auth w/ Okta OIDC remains functional for all users

**Screenshots**
![image](https://user-images.githubusercontent.com/16881666/180033713-ff1a19bb-04e3-4945-b172-68003ad72a27.png)

**Version**
2.4.7

**Logs**

```
argocd-server-5c58f5c844-cksjw argocd-server time="2022-07-20T15:55:59Z" level=warning msg="Failed to verify app state - got invalid redirectURL \"/\""
```

**Comment by crenshaw-dev** [comment]
@jmmclean have you been able to validate that the fix works for you?

**Comment by Mikerium** [comment]
+1 also experiencing this with dex configured for oidc with okta

**Comment by jmmclean** [comment]
@crenshaw-dev I will try to test it out today....TBH though I'm not sure how to reproduce (even tho i created the ticket). I think just hitting the argo-cd base href would show the issue (can you confirm)? I dont feel great reproducing the error, then quickly slinging the fix you have bc it impacts our dev environment...but ill talk w/ my team

**Comment by jmmclean** [comment]
@crenshaw-dev I rebuilt the image from your feature branch and tested, the login is now working.

Steps taken:
1. upgrade to v2.4.7
2. reproduced issue
3. upgrade only argocd-server to `image: carbonnexus/argocd:2.4.8-rc1`
4. could no longer reproduce the issue

**Comment by crenshaw-dev** [comment]
Fix released with 2.3.7 and 2.4.8. https://github.com/argoproj/argo-cd/pull/10116
